maro/simulator/scenarios/vm_scheduling/physical_machine.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Set
+
+from maro.backends.frame import NodeAttribute, NodeBase, node
+
+
+@node("pms")
+class PhysicalMachine(NodeBase):
+    """Physical machine node definition in frame."""
+    # Initial parameters.
+    id = NodeAttribute("i")
+    cpu_cores_capacity = NodeAttribute("i2")
+    memory_capacity = NodeAttribute("i2")
+    # Statistical features.
+    cpu_cores_allocated = NodeAttribute("i2")
+    memory_allocated = NodeAttribute("i2")
+
+    cpu_utilization = NodeAttribute("f")
+    energy_consumption = NodeAttribute("f")
+
+    def __init__(self):
+        """Internal use for reset."""
+        self._id = 0
+        self._init_cpu_cores_capacity = 0
+        self._init_memory_capacity = 0
+        # PM resource.
+        self._live_vms: Set[int] = set()
+
+    def set_cpu_utilization(self, cpu_utilization: float):
+        self.cpu_utilization = max(0, cpu_utilization)
```

Comment:
When the over-subscription happens, the cpu_utilization might be bigger than 100?

Reply:
I agree. How about adjusting the precision of the ``cpu_utilization`` in the end of ``set_cpu_utilization``? to prevent the misjudgment of over-load. like: (100 * 1 + 66.67 * 3) / 3 > 100

Reply:
Currently set it as 2 decimals. Not sure if we need less decimals? Like it may appears 100.03? or smth..

Reply:
As we don't support over-load now, it may be OK if the calculation result slightly bigger than 100.

When the over-load logic is enabled, we should carefully check this value. how about adding a hyper-parameter named **epsilon** to the over-load condition? like:
- epsilon = 0.5
- if PM.utilization > 100 + epsilon, then we think over-load happens; else we treat it as a precision issue.